### PR TITLE
feat(dismissible-box): allows a custom data role to close button and translations to aria label - FE-6309

### LINKED ISSUE #6487: Dismissible Box component should have a provision to add data-role to the close Icon
### Desired behaviour

Dismissible Box component should have a provision to add data-role to the close Icon

### Current behaviour

There is no provision to add data-role to the close Icon for tracking purposes

### Suggested Solution

_No response_

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/docs/dismissible-box--default-light-variant

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by DipperTheDan**
This change allows a custom data-role to be applied to the close icon button and ensures that the aria-label can be translated via locales.

fixes #6487

### Proposed behaviour

- Allow a custom `data-role` to be specified as part of the new `closeButtonDataProps` prop bag. 
- Adds translation support to `aria-label`

### Current behaviour

-  No `data-role` is specified
- No translation support for `aria-label`

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions

Check that `data-role` and `data-element` props can be passed to component using the test story. I have added `closeButtonDataProps` to the controls and they take an object.

### CODE REVIEW

File: src/components/dismissible-box/dismissible-box.pw.tsx [hunk 1]
```diff
@@ -98,6 +101,38 @@ test.describe("DismissibleBox component", () => {
       });
     }
   );
+
+  test("should have the expected data-role attribute on the IconButton, when a custom data-role is provided", async ({
```
[the remark this answers is not in the document]
Comment (DipperTheDan) [reply src/components/dismissible-box/dismissible-box.pw.tsx:105]: I agree that a unit test would suffice really. I only added this as we have done elsewhere so to keep things consistent. I'll happily delete these though. Less PW tests to maintain for us all 😆

File: src/components/dismissible-box/dismissible-box.component.tsx [hunk 2]
```diff
@@ -30,23 +34,36 @@ export interface DismissibleBoxProps
...
+      {...rest}
+    >
+      {children}
+      <span data-element="close-button-wrapper">
+        <IconButton
+          onClick={onClose}
+          aria-label={locale.dismissibleBox.ariaLabels.close()}
+          ml={3}
+          {...tagComponent("close-button", {
+            "data-element": "close-button",
```
[the remark this answers is not in the document]
Comment (DipperTheDan) [reply src/components/dismissible-box/dismissible-box.component.tsx:58]: I'm easy either way. We'll see what a second reviewer says and I'll go with the majority opinion.

File: src/locales/locale.ts [hunk 3]
```diff
@@ -46,6 +46,11 @@ interface Locale {
       close: () => string;
     };
   };
+  dismissibleBox: {
+    ariaLabels: {
```
[the remark this answers is not in the document]
Comment (DipperTheDan) [reply src/locales/locale.ts:50]: I did have it that way originally but when I saw that `dialog` and `dialogFullScreen` have done the aria-label for the close button the same way, I went for consistency.

[the remark this answers is not in the document]
Comment (tomdavies73) [reply src/locales/locale.ts:50]: Yeah, fair point @DipperTheDan, I'm all for being consistent, but I'd still prefer we go singular here but happy to listen to what the second reviewer thinks

File: src/components/dismissible-box/dismissible-box.component.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/dismissible-box/dismissible-box.component.tsx:58]: Yeah good spot Tom, I think we're better off using `Omit` here, it's what I did for [Time](https://github.com/Sage/carbon/blob/27dc669de37dab9d0549a5405b8eab17ef473121/src/components/time/time.component.tsx). Them passing data-component won't do anything but they won't get the feedback that it's an invalid prop etc

File: src/components/dismissible-box/dismissible-box.pw.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/dismissible-box/dismissible-box.pw.tsx:105]: I agree but also it's not a big deal to leave it now you've added it

[the remark this answers is not in the document]
Comment (DipperTheDan) [reply src/components/dismissible-box/dismissible-box.pw.tsx:105]: I'll just leave as is then if it's not doing any harm.

**Review by tomdavies73 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**